galaxy/api/views/survey.py

```diff
@@ -0,0 +1,126 @@
+# (c) 2012-2018, Ansible by Red Hat
+#
+# This file is part of Ansible Galaxy
+#
+# Ansible Galaxy is free software: you can redistribute it and/or modify
+# it under the terms of the Apache License as published by
+# the Apache Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# Ansible Galaxy is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# Apache License for more details.
+#
+# You should have received a copy of the Apache License
+# along with Galaxy.  If not, see <http://www.apache.org/licenses/>.
+
+import logging
+
+from galaxy.main import models
+from galaxy.api import serializers
+from . import base_views
+from rest_framework import status
+
+from rest_framework.exceptions import (APIException)
+from django.core.exceptions import ObjectDoesNotExist
+
+from rest_framework.response import Response
+
+logger = logging.getLogger(__name__)
+
+__all__ = [
+    'CommunitySurveyList',
+    'CommunitySurveyDetail'
+]
+
+SURVEY_FIElDS = (
+    'docs',
+    'ease_of_use',
+    'does_what_it_says',
+    'works_as_is',
+    'used_in_production'
+)
+
+
+class CommunitySurveyList(base_views.ListCreateAPIView):
+    model = models.CommunitySurvey
+    serializer_class = serializers.CommunitySurveySerializer
+
+    def post(self, request, *args, **kwargs):
+        data = request.data
+
+        try:
+            repo = models.Repository.objects.get(pk=data['repository'])
+        except ObjectDoesNotExist:
+            raise APIException(
+                'Repository does not exist.'
+            )
+
+        data['repository'] = repo
+        data['user'] = request.user
+
+        try:
+            survey = models.CommunitySurvey.objects.create(**data)
+        except Exception as exc:
+            raise APIException(
+                'Error creating survey: {0}'.format(exc.message)
+            )
+
+        serializer = self.get_serializer(instance=survey)
+
+        update_community_score(survey.repository)
+
+        return Response(serializer.data, status=status.HTTP_201_CREATED)
+
+
+class CommunitySurveyDetail(base_views.RetrieveUpdateDestroyAPIView):
+    model = models.CommunitySurvey
+    serializer_class = serializers.CommunitySurveySerializer
+
+    def update(self, request, *args, **kwargs):
+        data = request.data
+        instance = self.get_object()
+
+        if (request.user.id != instance.user.id):
+            return Response(status=status.HTTP_403_FORBIDDEN)
+
+        for key in data:
+            if key in SURVEY_FIElDS:
+                setattr(instance, key, data[key])
```

Comment:
Same as above. We need to start following DRF, and use proper validation techniques, as much as possible.